src/Microsoft.SqlTools.ServiceLayer/SchemaCompare/SchemaCompareOpenScmpOperation.cs

```diff
@@ -127,17 +129,38 @@ private SchemaCompareEndpointInfo GetEndpointInfo(bool source, SchemaCompareEndp
                         {
                             if (node.Parent.Name.ToString().Contains(searchingFor))
                             {
-                                endpointInfo.ConnectionDetails = SchemaCompareService.ConnectionServiceInstance.ParseConnectionString(node.Value);
-                                endpointInfo.ConnectionDetails.ConnectionString = node.Value;
-                                endpointInfo.DatabaseName = endpointInfo.ConnectionDetails.DatabaseName;
-                                endpointInfo.EndpointType = SchemaCompareEndpointType.Database;
+                                if(isProjectEndpoint)
```

Comment:
it's a little hard to follow this with both the project and database endpoint logic mixed together. Can they be split up so it's easier to follow and if there's some common code, that could be split into a helper function?

Reply:
Split- done.